src/context/TransactionsProvider/TransactionsProvider.tsx

```diff
@@ -55,6 +58,31 @@ export const TransactionsProvider: React.FC<TransactionsProviderProps> = ({ chil
 
       const { getOpportunitiesUserData } = opportunitiesApi.endpoints
 
+      const idleContractAddresses = [
+        ...Object.values(stakingOpportunitiesById)
+          .map(opportunity => (opportunity as IdleStakingSpecificMetadata | undefined)?.cdoAddress)
+          .filter(isSome),
+        IDLE_PROXY_1_CONTRACT_ADDRESS,
+        IDLE_PROXY_2_CONTRACT_ADDRESS,
+      ]
+      if (
+        transfers.some(
+          ({ from, to }) =>
+            idleContractAddresses.includes(from) || idleContractAddresses.includes(to),
+        )
+      ) {
+        dispatch(
+          getOpportunitiesUserData.initiate(
+            {
+              accountId,
+              defiType: DefiType.Staking,
+              defiProvider: DefiProvider.Idle,
+              opportunityType: DefiType.Staking,
+            },
+            { forceRefetch: true },
+          ),
+        )
+      }
       if (data?.parser === 'staking') {
```

Comment:
This should probably be an `else if` so the control flow clearly indicates that this dispatch is never called twice.

Reply:
05d16675cbf1c31cb156a0d97cba30b3d6b05032